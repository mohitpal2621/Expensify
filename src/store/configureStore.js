import { combineReducers, configureStore } from "@reduxjs/toolkit";
import expenseReducer from '../reducers/expenseReducer';
import filtersReducer from '../reducers/filterReducer';


const rootreducer = combineReducers({
    expenses: expenseReducer,
    filters: filtersReducer
})

export default () => {
    const store = configureStore({
        reducer: rootreducer
    });

    return store;
};
