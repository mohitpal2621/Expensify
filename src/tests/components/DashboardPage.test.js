// DashboardPage.test.js
import React from "react";
import { render, screen } from '@testing-library/react';
import ExpenseDashboardPage from "../../components/ExpenseDashboard";
import { MemoryRouter } from "react-router-dom";
import { Provider } from "react-redux";
import configureStore from "../../store/configureStore";

// This line of code is using Jest's jest.mock() function to mock the ExpenseListFilters component. 
// When you provide a module path to jest.mock(), Jest replaces the module with a mock implementation. 
// Here, we're providing a manual mock implementation directly inline with jest.mock().

// The () => () => <div>Mock ExpenseListFilters</div> function is an arrow function that returns another arrow function. 
// This inner arrow function is a functional component that simply returns a <div> element with the text "Mock ExpenseListFilters". 
//This is a mock implementation of the ExpenseListFilters component.

jest.mock("../../components/ExpenseListFilters", () => () => <div>Mock ExpenseListFilters</div>);
jest.mock("../../components/ExpenseList", () => () => <div>Mock ExpenseList</div>);

test("should render the dashboard page", () => {
    const store = configureStore();
    const {container} = render(
        <Provider store={store}>
            <MemoryRouter>
                <ExpenseDashboardPage/>
            </MemoryRouter>
        </Provider>
    );
    expect(container).toMatchSnapshot();
});
