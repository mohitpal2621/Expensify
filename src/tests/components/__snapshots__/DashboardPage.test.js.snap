// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the dashboard page 1`] = `
<div>
  <div>
    <div>
      Mock ExpenseListFilters
    </div>
    <div>
      Mock ExpenseList
    </div>
  </div>
</div>
`;
